File: play-interop-test/src/main/scala/controllers/GreeterServiceImpl.scala
//package controllers
//
//import akka.NotUsed
//import akka.stream.Materializer
//import akka.stream.scaladsl.Source
//import example.myapp.helloworld.grpc.helloworld.{ GreeterService, HelloReply, HelloRequest }
//import javax.inject.{ Inject, Singleton }
//
//import scala.concurrent.Future
//
///** Would be written by the user, with support for dependency injection etc */
//@Singleton
//class GreeterServiceImpl @Inject() (implicit mat: Materializer) extends GreeterService {
//
//  override def sayHello(in: HelloRequest): Future[HelloReply] = Future.successful(HelloReply(s"Hello, ${in.name}!"))
//
//  override def itKeepsTalking(in: Source[HelloRequest, NotUsed]): Future[HelloReply] = ???
//
//  override def itKeepsReplying(in: HelloRequest): Source[HelloReply, NotUsed] = ???
//
//  override def streamHellos(in: Source[HelloRequest, NotUsed]): Source[HelloReply, NotUsed] = ???
//}

File: play-interop-test/src/main/java/controllers/GreeterServiceImpl.java
package controllers;

import akka.NotUsed;
import akka.stream.javadsl.Source;
import example.myapp.helloworld.grpc.GreeterService;
import example.myapp.helloworld.grpc.HelloReply;
import example.myapp.helloworld.grpc.HelloRequest;

import java.util.concurrent.CompletionStage;

public class GreeterServiceImpl implements GreeterService {

  @Override
  public CompletionStage<HelloReply> sayHello(HelloRequest in) {
    return null;
  }

  @Override
  public CompletionStage<HelloReply> itKeepsTalking(Source<HelloRequest, NotUsed> in) {
    return null;
  }

  @Override
  public Source<HelloReply, NotUsed> itKeepsReplying(HelloRequest in) {
    return null;
  }

  @Override
  public Source<HelloReply, NotUsed> streamHellos(Source<HelloRequest, NotUsed> in) {
    return null;
  }
}
